modules/self-managed-node-group/main.tf

```diff
@@ -144,8 +149,20 @@ resource "aws_launch_template" "this" {
     }
   }
 
-  iam_instance_profile {
-    arn = var.create_iam_instance_profile ? aws_iam_instance_profile.this[0].arn : var.iam_instance_profile_arn
+  dynamic "iam_instance_profile" {
+    for_each = var.create_iam_instance_profile || var.iam_instance_profile_arn != null ? [1] : []
+
+    content {
+      arn = var.create_iam_instance_profile ? aws_iam_instance_profile.this[0].arn : var.iam_instance_profile_arn
+    }
+  }
+
+  dynamic "iam_instance_profile" {
+    for_each = !var.create_iam_instance_profile && var.iam_role_name != null ? [1] : []
+
+    content {
+      name = var.iam_role_name
+    }
```

Comment:
I'd like to see a rationale for this change as well

Reply:
From our internal PR:
>allow passing along IAM roles for node group instead of creating a new one

Reply:
I don't have much context on this particular change since it was done by another folk.
This change was picked up from our internal repo, namely:
> Improve support for self-managed node groups `#29`

I will circle back with the author to provide more information on this.

Reply:
From what I can see in the code here, I think we assume an instance profile already exists whose name matches the role the instance profile is using.

I think we could end up in an error condition (two iam_instance_profile blocks) if both `var.iam_instance_profile_arn` and `var.iam_role_name` are set with `var.create_iam_instance_profile == false`

Reply:
I think fine to keep this as is, but we should probably create a followup PR to handle this case ^

Reply:
Agree, it's fine for now and we can revise this later to eliminate possible conflict.

Reply:
Fixed the logic in https://github.com/pinterest/terraform-aws-eks/pull/3/commits/dc33cfcfef2a996fd41ade56cf78571011e0f313